python/ray/data/_internal/datasource/clickhouse_datasink.py

```diff
@@ -0,0 +1,368 @@
+            if result and result.result_rows:
+                return result.result_rows[0][0] == 1
+            return False
+        except Exception as e:
+            logger.warning(f"Could not verify if table {self._table} exists: {e}")
+            return False
+
+    def _get_existing_order_by(self, client) -> Optional[str]:
+        logger.debug(
+            f"Retrieving ORDER BY clause from SHOW CREATE TABLE for {self._table}"
+        )
+        try:
+            show_create_sql = self._SHOW_CREATE_TABLE_TEMPLATE.format(
+                table_name=self._table
+            )
+            result = client.command(show_create_sql)
+            ddl_str = str(result)
+            pattern = r"(?is)\border\s+by\s+(.*?)(?=\bengine\b|$)"
+            match = re.search(pattern, ddl_str)
+            if match:
+                return match.group(1).strip()
+            return None
+
+        except Exception as e:
+            logger.warning(
+                f"Could not retrieve SHOW CREATE TABLE for {self._table}: {e}"
+            )
+            return None
+
+    @property
+    def supports_distributed_writes(self) -> bool:
+        return True
+
+    def on_write_start(self) -> None:
+        client = None
+        try:
+            client = self._init_client()
+            table_exists = self._table_exists(client)
+            if self._mode == SinkMode.OVERWRITE:
+                # Drop table if it exists.
+                drop_sql = self._DROP_TABLE_TEMPLATE.format(table_name=self._table)
+                logger.info(f"Mode=OVERWRITE => {drop_sql}")
+                client.command(drop_sql)
+                self._table_dropped = True
+                # If the old table existed and no explicit "order_by" is set,
+                # adopt the existing ORDER BY from the old table DDL.
+                if table_exists and self._table_settings.get("order_by") is None:
+                    existing_order_by = self._get_existing_order_by(client)
+                    if existing_order_by is not None:
+                        self._table_settings["order_by"] = existing_order_by
+                        logger.info(
+                            f"Reusing old ORDER BY from overwritten table: {existing_order_by}"
+                        )
+            elif self._mode == SinkMode.CREATE:
+                if table_exists:
+                    msg = (
+                        f"Table {self._table} already exists in mode='CREATE'. "
+                        "Use mode='APPEND' or 'OVERWRITE' instead."
+                    )
+                    logger.error(msg)
+                    raise RuntimeError(msg)
+            elif self._mode == SinkMode.APPEND:
+                if table_exists and self._table_settings.get("order_by") is None:
+                    existing_order_by = self._get_existing_order_by(client)
+                    if existing_order_by:
+                        self._table_settings["order_by"] = existing_order_by
```

Comment:
I'd suggest to report an error if these 2 diverge, rather than just silently overwriting it